plugins/bulk-import/src/components/PreviewFile/PreviewPullRequestForm.tsx

```diff
@@ -271,31 +215,46 @@ export const PreviewPullRequestForm = ({
       label: 'Annotations',
       name: 'prAnnotations',
       value:
-        pullRequest[repoId]?.prAnnotations ??
+        formik.values[repoId]?.prAnnotations ??
         convertKeyValuePairsToString(
-          pullRequest[repoId]?.yaml?.metadata?.annotations,
+          formik.values[repoId]?.yaml?.metadata?.annotations,
         ),
     },
     {
       label: 'Labels',
       name: 'prLabels',
       value:
-        pullRequest[repoId]?.prLabels ??
+        formik.values[repoId]?.prLabels ??
         convertKeyValuePairsToString(
-          pullRequest[repoId]?.yaml?.metadata?.labels,
+          formik.values[repoId]?.yaml?.metadata?.labels,
         ),
     },
     {
       label: 'Spec',
       name: 'prSpec',
       value:
-        pullRequest[repoId]?.prSpec ??
+        formik.values[repoId]?.prSpec ??
         convertKeyValuePairsToString(
-          pullRequest[repoId]?.yaml?.spec as Record<string, string>,
+          formik.values[repoId]?.yaml?.spec as Record<string, string>,
         ),
     },
   ];
 
+  React.useEffect(() => {
+    const err = {
+      ...formErrors,
+      [repoId]: formik.errors[repoId] as any as PullRequestPreview,
+    };
+
+    if (!formik.errors[repoId]) {
+      delete err[repoId];
+    }
+
+    setFormErrors(err);
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [formik.errors]);
```

Comment:
This tries to pass the errors from the sub-form into the parent form? Why?

Reply:
The sub-form can be a single preview form or a tabbed view of multiple preview forms and the save button is the common button in the main form that saves all the changes made in the sub form(s)